applications/gui/modules/validators.c

```diff
@@ -26,12 +34,15 @@ bool validator_is_file_callback(const char* text, string_t error, void* context)
     return ret;
 }
 
-ValidatorIsFile*
-    validator_is_file_alloc_init(const char* app_path_folder, const char* app_extension) {
+ValidatorIsFile* validator_is_file_alloc_init(
+    const char* app_path_folder,
+    const char* app_extension,
+    const char* current_name) {
```

Comment:
There is one case that is going to break validator logic: text_input and current_name same buffer. It's better to strdup and then free on validator release